weatherHome.js

```diff
@@ -1,38 +1,192 @@
   document.querySelector(".humidity").innerHTML = data.main.humidity + "%";
+  iconGeneration(data.weather[0].icon).then((imageUrl) => {
+    document.getElementById("tempIcon").src = imageUrl;
+  });
+}
+
+async function displayWeekWeather() {
+  const apikey = "8bbf28e0cbcb834f5a430d84e87a32e4";
+  const apiUrl =
+    "https://api.openweathermap.org/data/2.5/forecast?lat=44.34&lon=10.99&units=metric";
+  const cityName = document.getElementById("userInput").value;
+  const response = await fetch(apiUrl + `&q=${cityName}&appid=${apikey}`);
+  const data = await response.json();
+  const week = [
+    data.list[0],
+    data.list[5],
+    data.list[12],
+    data.list[20],
+    data.list[30],
+    data.list[36],
+  ];
+  document.getElementById("weekCol").innerHTML = "";
+  for (let i in week) {
+    const pElement = document.createElement("p");
+    pElement.textContent = Math.round(week[i].main.temp) + "°C";
+    const imgElement = document.createElement("img");
+    const imageUrl = await iconGeneration(week[i].weather[0].icon);
+    imgElement.src = imageUrl;
+    const divElement = document.createElement("div");
+    const h4Element = document.createElement("h4");
+    h4Element.textContent = getDate(week[i].dt_txt);
+    divElement.appendChild(h4Element);
+    divElement.appendChild(imgElement);
+    divElement.appendChild(pElement);
+    document.getElementById("weekCol").appendChild(divElement);
+  }
+}
+
+function windDisplayFunction(data) {
+  const windColElement = document.getElementById("windCol");
+  const windDegree = data.wind.deg;
+  document.getElementById("windDegree").innerHTML = windDegree + "°";
+  let wind = "";
+
+  if (windDegree >= 0 && windDegree < 22.5) {
+    wind = "North wind";
+  } else if (windDegree >= 22.5 && windDegree < 45) {
+    wind = "North wind";
+  } else if (windDegree >= 45 && windDegree < 67.5) {
+    wind = "Northeast wind";
+  } else if (windDegree >= 67.5 && windDegree < 90) {
+    wind = " East-northeast wind";
+  } else if (windDegree >= 90 && windDegree < 112.5) {
+    wind = "East wind";
+  } else if (windDegree >= 112.5 && windDegree < 135) {
+    wind = "East-southeast wind";
+  } else if (windDegree >= 135 && windDegree < 157.5) {
+    wind = "Southeast wind";
+  } else if (windDegree >= 157.5 && windDegree < 180) {
+    wind = "South-southeast wind";
+  } else if (windDegree >= 180 && windDegree < 225) {
+    wind = "South wind";
+  } else if (windDegree >= 225 && windDegree < 270) {
+    wind = "Southwest wind";
+  } else if (windDegree >= 270 && windDegree < 315) {
+    wind = "West wind";
+  } else if (windDegree >= 315 && windDegree < 360) {
+    wind = "Southeast wind";
+  } else {
+    wind = "North Wind";
+  }
+  document.getElementById("windName").innerHTML = wind;
+}
+
+function sunDisplay(data) {
+  const sunRiseTimeUnixValue = data.sys.sunrise;
+  const riseDate = new Date(sunRiseTimeUnixValue * 1000);
+  const riseHours = ("0" + riseDate.getHours()).slice(-2);
```

Comment:
Simplify this function, create a formatTime function and move the common code to it